src/ngraph/op/tensor_iterator.cpp

```diff
@@ -0,0 +1,128 @@
+    {
+        size_t input_index = input.get_index();
+        Output<Node> value = input.get_source_output();
+        PartialShape sequence_shape = value.get_partial_shape();
+        PartialShape iterator_shape = sequence_shape;
+        Rank sequence_rank = sequence_shape.rank();
+
+        if (sequence_rank.is_dynamic())
+        {
+            // Can't determine the sequence length
+            iteration_count_dynamic = true;
+        }
+        else
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  static_cast<size_t>(sequence_shape.rank()) != 0,
+                                  "Input ",
+                                  input_index,
+                                  " is specified to be a sequence but is scalar.");
+            Dimension sequence_dim = sequence_shape[0];
+            vector<Dimension> dimensions = static_cast<vector<Dimension>>(sequence_shape);
+            dimensions.erase(dimensions.begin());
+            iterator_shape = PartialShape(dimensions);
+
+            if (sequence_dim.is_dynamic())
+            {
+                // Can't determine the sequence length
+                iteration_count_dynamic = true;
+            }
+            else
+            {
+                size_t sequence_length = static_cast<size_t>(sequence_dim);
+                if (!iteration_count_valid || (sequence_length < iteration_count))
+                {
+                    iteration_count = sequence_length;
+                    iteration_count_valid = true;
+                }
+            }
+        }
+
+        NODE_VALIDATION_CHECK(
+            this,
+            iterator_shape.compatible(m_body_parameters.at(input_index)->get_partial_shape()),
+            "Iterator body param is not compatible with value");
+    }
+    // The body may depend on the body parameters as well as values from outside the body
+    // Body parameters depend on the loop initialization
+    NodeVector body_result_nodes;
+    for (auto& body_output : m_body_outputs)
+    {
+        body_result_nodes.push_back(body_output.get_node_shared_ptr());
+    }
+    std::list<std::shared_ptr<Node>> body_node_closure(topological_sort(body_result_nodes, true));
+    std::set<Node*> bound_nodes;
+    std::vector<Node*> free_nodes;
+    for (auto& parameter : m_body_parameters)
+    {
+        std::cerr << *this << " Bound: " << *parameter << std::endl;
+        bound_nodes.insert(parameter.get());
+    }
+    for (auto& node : body_node_closure)
+    {
+        if (bound_nodes.find(node.get()) == bound_nodes.end())
+        {
+            bool is_free = true;
+            for (auto input : node->inputs())
+            {
+                auto input_node = input.get_source_output().get_node();
+                if (bound_nodes.find(input_node) != bound_nodes.end())
+                {
+                    bound_nodes.insert(node.get());
+                    is_free = false;
+                    std::cerr << *this << " Bound: "
```

Comment:
Why do you use `std::cerr` here? Is it really unsupported case or it is just debug information? 
If it is unsupported case, assert is better solution.

Reply:
This is just debugging.